lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -2999,30 +2999,39 @@ ThreadContext::ClearScriptContextCaches()
 void
 ThreadContext::ClearInlineCachesWithDeadWeakRefs()
 {
+#ifdef ENABLE_JS_ETW
+    size_t allocatedSize = 0;
+    size_t preClearFreeListSize = 0;
+    size_t freeListSize = 0;
+    size_t polyInlineCacheSize = 0;
+    uint scriptContextCount = 0;
+
+    // Note: this event is not meaningful for MemGC, only Chakra
+    JS_ETW(EventWriteJSCRIPT_GC_CLEAR_INLINECACHE_START());
+#endif
+
     for (Js::ScriptContext *scriptContext = scriptContextList; scriptContext != nullptr; scriptContext = scriptContext->next)
     {
+#ifdef ENABLE_JS_ETW
+        scriptContextCount++;
+        allocatedSize += scriptContext->GetInlineCacheAllocator()->AllocatedSize();
+        preClearFreeListSize += scriptContext->GetInlineCacheAllocator()->FreeListSize();
+#endif
+
         scriptContext->ClearInlineCachesWithDeadWeakRefs();
+
+#ifdef ENABLE_JS_ETW
+        freeListSize += scriptContext->GetInlineCacheAllocator()->FreeListSize();;
+        polyInlineCacheSize += scriptContext->GetInlineCacheAllocator()->GetPolyInlineCacheSize();
+#endif
     }
 
+    JS_ETW(EventWriteJSCRIPT_GC_CLEAR_INLINECACHE_STOP(this, scriptContextCount, (uint) allocatedSize, (uint) preClearFreeListSize, (uint) freeListSize, (uint) polyInlineCacheSize));
+
     if (PHASE_TRACE1(Js::InlineCachePhase))
     {
-        size_t size = 0;
-        size_t freeListSize = 0;
-        size_t polyInlineCacheSize = 0;
-        uint scriptContextCount = 0;
-        for (Js::ScriptContext *scriptContext = scriptContextList;
-            scriptContext;
-            scriptContext = scriptContext->next)
-        {
-            scriptContextCount++;
-            size += scriptContext->GetInlineCacheAllocator()->AllocatedSize();
-            freeListSize += scriptContext->GetInlineCacheAllocator()->FreeListSize();
-#ifdef POLY_INLINE_CACHE_SIZE_STATS
-            polyInlineCacheSize += scriptContext->GetInlineCacheAllocator()->GetPolyInlineCacheSize();
-#endif
-        };
         Output::Print(_u("Inline cache arena: total = %5I64u KB, free list = %5I64u KB, poly caches = %5I64u KB, script contexts = %u\n"),
-            static_cast<uint64>(size / 1024), static_cast<uint64>(freeListSize / 1024), static_cast<uint64>(polyInlineCacheSize / 1024), scriptContextCount);
+            static_cast<uint64>(allocatedSize / 1024), static_cast<uint64>(freeListSize / 1024), static_cast<uint64>(polyInlineCacheSize / 1024), scriptContextCount);
```

Comment:
This needs to be under `#ifdef ENABLE_JS_ETW` or you need the `allocatedSize` declaration to be outside of it.